DataDog/datadog-ci-rb
# [CIVIS-9951] add settings option to ignore code coverage for bundled gems location

**What does this PR do?**
In CI environments bundled gems are often installed inside the project folder for caching purposes. Unfortunately, it makes per test code coverage for ITR extremely slow as we need to unnecessary track all the code files from the external gems.

This PR adds `DD_CIVISIBILITY_ITR_CODE_COVERAGE_EXCLUDED_BUNDLE_PATH` environment variable that allows us to set a certain path to be excluded from code coverage (supports both relative and absolute paths). Even better, for the most cases users won't have to use this configuration: we will locate bundle location and set it for them.

It works as following:
- if `bundler` gem is available in the environment, we use `Bundler.bundle_path` and set this result as ignored path
- otherwise we try to locate `vendor/bundle` or `.bundle` under the git root - if any of them exist, we ignore them

**How to test the change?**
Unit tests are provided. 

I have run test-environment with gems bundled inside the project directory to confirm the fix:
https://github.com/DataDog/test-environment/pull/284

The overhead with bundling gems in the project folder in this case is insignificantly larger: 113% vs 106% for rubocop for example. Without this change, it would be 450% vs 106%.
